lib/symboldatabase.cpp

```diff
@@ -4350,6 +4350,18 @@ const Function *Function::getOverriddenFunction(bool *foundAllBaseClasses) const
     return getOverriddenFunctionRecursive(nestedIn->definedType, foundAllBaseClasses);
 }
 
+// prevent recursion if base is the same except for different template parameters
+static bool isDerivedFromItself(const std::string& thisName, const std::string& baseName)
+{
+    const auto pos = thisName.find('<');
```

Comment:
I think that if this a hot path we could add something like this first to get a quick bailout:

    if (thisName.back() != '>')
        return false;

Well if it's not hot then it's just extra code so feel free to skip that.